grails-app/services/au/org/ala/bie/WordpressService.groovy

```diff
@@ -42,13 +42,38 @@ class WordpressService implements IndexingInterface {
      * @param url The site map url
      * @return
      */
-    private List crawlWordPressSite(URL url) throws Exception {
+    private List crawlWordPressSite(Queue<URL> queue) throws Exception {
+        Set locations = [] as Set
+        Set<URL> seen = [] as Set
         // get list of pages to crawl via Google sitemap xml file
         // Note: sitemap.xml files can be nested, so code may need to read multiple files in the future (recursive function needed)
-        Document doc = Jsoup.connect(url.toExternalForm()).get()
-        Elements pages = doc.select("loc")
-        log.info("Sitemap file lists " + pages.size() + " pages.")
-        return pages.collect { it.text() }
+        while (!queue.isEmpty()) {
+            URL map = queue.remove()
```

Comment:
variable name of `map` is slightly confusing being of type `URL` -  would be nicer to name this in a more specific way. `seen` and `map` have a relationship so having a common naming 'part' would help readability (e.g. `url` and `seenUrls`).

Reply:
I've fixed this and then got a little carried away ...